apps/devportal/data/markdown/pages/learn/accelerate/xm-cloud/pre-development/developer-experience/dev-workflow-codespaces.md

```diff
@@ -0,0 +1,201 @@
+      "version": "lts",
+      "nvmVersion": "latest"
+    }
+  },
+
+  // Use 'forwardPorts' to make a list of ports inside the container available locally.
+  "forwardPorts": [3000],
+  "portsAttributes": {
+    "3000": {
+      "visibility": "public"
+    }
+  }
+}
+```
+
+In this file, we are telling Codespaces:
+
+- The base image to use for the development container. For XM Cloud, we will use the dotnet v6 base image. This is to support the Sitecore dotnet tool CLI. When the tool is updated to use dotnet 7.x, the base image can be updated.
+- Next we add Node LTS in as a feature to be able to run the Next.js application.
+- Finally we set port 3000 up for port forwarding for the Codespaces URL and we make port 3000 public so that it can be accessed by Pages.
+
+Commit these changes to your git repository and move on to the next step, based on where your git repository is hosted.
+
+### Setting up your Codespace - Github
+
+Setting up your codespace on github is very simple. Navigate to the code page in your repository and click the green Code button on the right. Change to the Codespaces tab and click the Create codespace on main button:
+
+<img src="/images/learn/accelerate/xm-cloud/codespaces-1.png" alt="Create Codespace on Main"/>
+
+This will open a new tab with your codespace being created. The configuration will include all the required resources to run your web application.
+
+The UI is the same as Visual Studio Code. Once the Codespace is created, the developer will be able to open that Codespace in Visual Studio Code also.
+
+### Setting up your Codespace - Azure DevOps
+
+Go to [Github Codespaces](https://github.com/codespaces) and use the Blank quick start template to create a new codespace. You can also do this within Visual Studio Code by installing the Visual Studio Codespaces plugin. This recipe will focus on using the browser, but the same steps apply to using VS Code.
+
+These steps assume you already have an XM Cloud repository based on the XM Cloud Foundation Head starter kit and have gone through the steps of adding the devcontainer.json file.
+
+Next, we need to initialize the codespace with a git repository. Open a terminal in the root of the codespace and run:
+
+```powershell
+git init
+```
+
+Now, we will add the Azure DevOps repository as a remote.
+
+```powershell
+git remote add origin git@ssh.dev.azure.com........
+```
+
+You can copy the SSH or HTTP git url from the repository’s website like this:
+
+<img src="/images/learn/accelerate/xm-cloud/codespaces-2.png" alt="Get Azure DevOps git repo url"/>
+
+Now run:
+
+```powershell
+git pull
+```
+
+This will pull the repository into the Codespace ready to run. Before running, the container will need to be rebuilt. In codespaces, bring up the command pallet and run `Codespaces: Rebuild Container`.
+
+### Running your website
+
+To run the website for the first time, first the developer needs to configure the environment variables for the solution.
```

Comment:
To run the website for the first time, first the developer... (remove the second "first")

Reply:
Cleaned up the language here.